core/strategies.go

```diff
@@ -7,11 +7,24 @@ import (
 
 // StrategyI defines
 type StrategyI interface {
+	// GetType returns the type of the strategy
 	GetType() string
+
+	// SetupRelay performs chain-specific setup for `src` and `dst` before starting the relay
 	SetupRelay(ctx context.Context, src, dst *ProvableChain) error
-	UnrelayedPackets(src, dst *ProvableChain, sh SyncHeaders) (*RelayPackets, error)
+
+	// UnrelayedPackets returns packets to execute RecvPacket to on `src` and `dst`.
+	// `includeRelayedButUnfinalized` decides if the result includes packets of which recvPacket has been executed but not finalized
+	UnrelayedPackets(src, dst *ProvableChain, sh SyncHeaders, includeRelayedButUnfinalized bool) (*RelayPackets, error)
+
+	// RelayPackets executes RecvPacket to the packets contained in `rp` on both chains (`src` and `dst`).
 	RelayPackets(src, dst *ProvableChain, rp *RelayPackets, sh SyncHeaders) error
-	UnrelayedAcknowledgements(src, dst *ProvableChain, sh SyncHeaders) (*RelayPackets, error)
+
+	// UnrelayedAcknowledgements returns packets to execute AcknowledgePacket to on `src` and `dst`.
+	// `includeRelayedButUnfinalized` decides if the result includes packets of which acknowledgePacket has been executed but not finalized
+	UnrelayedAcknowledgements(src, dst *ProvableChain, sh SyncHeaders, includeRelayedButUnfinalized bool) (*RelayPackets, error)
+
+	// RelayAcknowledgements executes AcknowledgePacket to the packets contained in `rp` on both chains (`src` and `dst`).
```

Comment:
which is right, `rp` or `ra`?

Reply:
Thank you. `ra` is correct.

Reply:
No, `rp` is correct!
`rp`, `ra`, and `sp` are mixed at the moment, but I will fix them all to `rp`.